addons/hr/models/hr_employee.py

```diff
@@ -125,11 +123,36 @@ class HrEmployeePrivate(models.Model):
         ('user_uniq', 'unique (user_id, company_id)', "A user cannot be linked to multiple employees in the same company.")
     ]
 
-    def _get_placeholder_filename(self, field=None):
-        image_fields = ['image_%s' % size for size in [1920, 1024, 512, 256, 128]]
-        if field in image_fields:
-            return 'hr/static/src/img/default_image.png'
-        return super()._get_placeholder_filename(field=field)
+    @api.depends('name', 'user_id.share', 'image_1920')
+    def _compute_avatar_1920(self):
+        super()._compute_avatar_1920()
+
+    @api.depends('name', 'user_id.share', 'image_1024')
+    def _compute_avatar_1024(self):
+        super()._compute_avatar_1024()
+
+    @api.depends('name', 'user_id.share', 'image_512')
+    def _compute_avatar_512(self):
+        super()._compute_avatar_512()
+
+    @api.depends('name', 'user_id.share', 'image_256')
+    def _compute_avatar_256(self):
+        super()._compute_avatar_256()
+
+    @api.depends('name', 'user_id.share', 'image_128')
+    def _compute_avatar_128(self):
+        super()._compute_avatar_128()
+
+    def _compute_avatar(self, avatar_field, image_field):
+        employees_with_internal_user = self.filtered(lambda employee: employee.user_id and not employee.user_id.share)
+        super(HrEmployeePrivate, employees_with_internal_user)._compute_avatar(avatar_field, image_field)
+        for employee in self - employees_with_internal_user:
+            employee[avatar_field] = employee[image_field] or employee._avatar_get_placeholder()
+
+    def _avatar_get_seed(self):
```

Comment:
According to me, unnecessary override as base seed seems ok

Reply:
The goal here is to have a match of color between employee and its user (if set). As the seed depends on `create_date`, and the employee and user can (and most likely will) be created at a different time, this is the reason of the override.

Reply:
Completely out of the blue ... you should take user's image if set, otherwise it makes no sense to duplicate images and have code just for that.

Reply:
We need to call `_avatar_get_seed` on res.user counterpart, not on the employee

Reply:
This makes your code unnecessarily complex. Simple code. Please.

Reply:
I agree in general, but this is the identity issue that we have so much trouble resolving. I don't think this PR is the right place to figure out how we want to share data (image in this case) between user, partner, employee, ...

The avatar mixin is not about syncing data or resolving identity, it's only about managing placeholder when the image (wherever it is coming from) is not set.

Reply:
Nope, you always send that kind of questions to "this is not the palce to discuss". This is the place to do it: your PR should be as minimal as possible. Her code takes way too much overrides just to generate a one-letter avatar. It should be contained in a single utility, not span accross 3 models... Keep it simple, keep it simple, keep it simple.

Identity does not come with an avatar color. It comes with naming or clarity in models.

Reply:
employee must have somewhere code that says "if user, take value from user", whether the override is in seed or main compute method, the override will need to exist. The only way to have absolutely no override is if all models have a related to identity which is the only place where the avatar is.

I'll push a change to keep override only in compute and remove the seed method, let me know if you like it better that way.

Reply:
Isn't it @leonardopavanrocha task ^^ ? Anyway I think it deserves some thoughts indeed, no need to rush, unless I am mistaken. But main purpose of my remarks are always: less code unless really necessary (aka, strong functional requirement), especially for a V1.

Reply:
I'm on Discord with him, but he's AFK for a bit, we're just working together to get it done for the freeze.